### DISCUSSION TITLE
Need some help with completion

### DISCUSSION THREAD

**Discussion by SebastienCFC** [icsharpcode/AvalonEdit#428] [Q&A] [answered]
Hello,

I am working on a WPF project using .NET 8.0 and Visual Studio 2022.

I would like to implement code completion functionality as follows:

First case: The user opens the completion window using Ctrl+Space, enters some text, and selects the desired item from the list. In this case, the user uses completion to enter an entire word.

Second case: In contrast to the first case, the user starts typing text before opening the completion window, and the completion window opens with suggestions based on the text that has already been entered. In this case, the user uses completion to complete an already started entry.

The first case works perfectly fine, but the second case is causing me problems.

I hope this is clear.

Sincerely,

**Comment by siegfriedpammer** [comment]
Would you mind explaining what problems you are experiencing? Also, showing parts of what you tried to do would be nice. Thanks

**Reply by SebastienCFC** [reply]
What i want : 
Behavior expected :  open the completion list already filtered, example : i type "pk", open the completion window manually (with ctrl + space), the list show me only items start with "pk". 
Additional behavior expected : when i continue typing new characters the list refresh the filter.

What i have :
i type "pk", open the completion window manually the list show all items (no filter), and select the first item start with "pk".
Additional : when i continue typing new characters the list refesh the filter, but don't take in count "pk", if type "a" after "pk" the list will be filtered all items start with "a" instead of "pka".

Sorry if it's not clear, i have some difficulty with english.

**Reply by siegfriedpammer** [reply, marked as answer]
Take a look at the StartOffset and EndOffset properties of the completion window.

**Comment by SebastienCFC** [comment]
It's exactly what i need.

Thank you.
